mlrun/runtimes/serving.py

```diff
@@ -374,16 +400,30 @@ def deploy(self, dashboard="", project="", tag="", verbose=False):
             # initialize or create required streams/queues
             self.spec.graph.check_and_process_graph()
             self.spec.graph.init_queues()
+
+        # Handle secret processing before handling child functions, since secrets are transferred to them
+        if self.spec.secret_sources:
+            # Before passing to remote builder, secrets values must be retrieved (for example from ENV)
+            # and stored as inline secrets. Otherwise, they will not be available to the builder.
+            self._secrets = SecretsStore.from_list(self.spec.secret_sources)
+            self.spec.secret_sources = self._secrets.to_serial()
+            # Add vault configurations to pod spec, if vault secret source was added.
+            if self._secrets.has_vault_source():
+                project_name = project or self.metadata.project
+                self._add_vault_params_to_spec(project=project_name)
```

Comment:
you're in the client context here, you can't call this

Reply:
You're correct. Will fix.